src/sentry/web/api.py

```diff
@@ -500,6 +502,63 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
         return event_id
 
 
+class MinidumpView(StoreView):
+    helper_cls = MinidumpApiHelper
+    content_types = ('multipart/form-data')
+
+    def _dispatch(self, request, helper, project_id=None, origin=None, *args, **kwargs):
+        # TODO(ja): Refactor shared code with CspReportView
+
+        # A minidump submission as implemented by Breakpad and Crashpad or any
+        # other library following the Mozilla Soccorro protocol is a POST request
+        # without Origin or Referer headers. Therefore, we cannot validate the
+        # origin of the request, but we *can* validate the "prod" key in future.
+        if request.method != 'POST':
+            return HttpResponseNotAllowed(['POST'])
+
+        content_type = request.META.get('CONTENT_TYPE')
+        if content_type is None or not content_type.startswith(self.content_types):
+            raise APIError('Invalid Content-Type')
+
+        request.user = AnonymousUser()
+
+        project = self._get_project_from_id(project_id)
+        helper.context.bind_project(project)
+        Raven.tags_context(helper.context.get_tags_context())
+
+        # This is yanking the auth from the querystring since it's not
+        # in the POST body. This means we expect a `sentry_key` and
+        # `sentry_version` to be set in querystring
+        auth = helper.auth_from_request(request)
```

Comment:
we should make this this work and add a test.

Reply:
Same goes for CSP below. Also, I noticed the comment (copy-pasted from CSP) is wrong, as `sentry_version` is not required in the querystring. It's not even checked anywhere. Would you like to see all this in this PR or a separate one?

Reply:
You can add tests later. Just leave a note somewhere to not forget.